compileable versions of ThePEG, JEWEL, CRMC:

qgp opened the pull request:
- fixes in deps (capitalization)
- adding JEWEL
- adding CRMC

dberzano commented:
Hello @qgp, could you change lhapdf5 -> lhapdf? We use Git branches to distinguish between versions, for instance we have the master branch using ROOT 5 and a "ROOT 6" branch with the same recipes but ROOT version 6. Thanks!

qgp commented:
The issue is depending on lhapdf v5, see alisw/alibuild#64 where @ktf suggested to use lhapdf and lhapdf5. The situtation is somewhat similar to pythia6 and pythia8. I can change this if there is another way to depend on a specific version of LHAPDF.

dberzano commented on crmc.sh:
```diff
@@ -0,0 +1,41 @@
+# Dependencies
+module load BASE/1.0 yaml-cpp/$YAML_CPP_VERSION-$YAML_CPP_REVISION boost/$BOOST_VERSION-$BOOST_REVISION autotools/$AUTOTOOLS_VERSION-$AUTOTOOLS_REVISION
+# Our environment
+setenv LHAPDF_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
Watch out, here it should be `CRMC_ROOT` instead of `LHAPDF_ROOT`, could you fix it?

dberzano commented on jewel.sh:
```diff
@@ -0,0 +1,44 @@
+module-whatis "ALICE Modulefile for $PKGNAME $PKGVERSION-@@PKGREVISION@$PKGHASH@@"
+# Dependencies
+module load BASE/1.0
+# Our environment
```
Here you should set `JEWEL_ROOT`.

dberzano commented on lhapdf5.sh:
```diff
@@ -0,0 +1,42 @@
+# Dependencies
+module load BASE/1.0 yaml-cpp/$YAML_CPP_VERSION-$YAML_CPP_REVISION boost/$BOOST_VERSION-$BOOST_REVISION autotools/$AUTOTOOLS_VERSION-$AUTOTOOLS_REVISION
+# Our environment
+setenv LHAPDF_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
```
Since the package is `lhapdf5`, then `LHAPDF_ROOT` -> `LHAPDF5_ROOT`.

dberzano commented on thepeg.sh:
```diff
@@ -6,29 +6,33 @@ requires:
+export LD_LIBRARY_PATH="${LHAPDF5_ROOT}/lib:${CGAL_ROOT}/lib:${BOOST_ROOT}/lib:${LD_LIBRARY_PATH}"
+export LIBRARY_PATH="${LHAPDF5_ROOT}/lib:${CGAL_ROOT}/lib:${BOOST_ROOT}/lib:${LIBRARY_PATH}"
+export LHAPATH="${LHAPDF5_ROOT}/share/lhapdf"
+export PERLLIB=/usr/share/perl5
```
The AliEn-Runtime package contains a version of perl5, but you can leave it like this, I'll take care of it.

qgp replied:
But then we have to depend on AliEn-Runtime, don't we? And alien is not really required here. So, this seems like a strange way to pull in perl.

qgp commented:
Thanks for the comments, I'll update the pull request accordingly.

qgp commented:
Hi @dberzano, I updated the pull request and now also changed the dependencies in the module files to consistently use $PKG_VERSION-$PKG_REVISION. I assume you prefer this rather than having hardcoded versions. Thanks!

ktf commented on thepeg.sh:
```diff
@@ -6,29 +6,33 @@ requires:
-  -name Makefile.in \
-  -exec touch '{}' \;
-autoreconf -ivf
+#find . \
```
Please try to avoid commenting code.

qgp replied:
Understood. Here, I left it in on purpose because Dario said he needed that part, while I did not now (on Ubuntu). And we might indeed need it for SLC 5 which we can't build and test yet because of the missing dependencies (as a result of the lack of python).
